Turf/Turf.swift

```diff
@@ -285,3 +288,67 @@ public struct Polyline {
         return closestCoordinate
     }
 }
+
+public struct Polygon {
+    var polygonCoordinates: [[CLLocationCoordinate2D]]
+    
+    var polygonArea: Double {
+        var area:Double = 0
+   
+        if (!polygonCoordinates.isEmpty && polygonCoordinates.count > 0) {
+            
+            area += abs(ringArea(polygonCoordinates[0]))
+
+            for coordinate in polygonCoordinates.suffix(from: 1) {
+                area -= abs(ringArea(coordinate))
+            }
+        }
+        return area
+    }
+    
+    /**
+     * Calculate the approximate area of the polygon were it projected onto the earth.
+     * Note that this area will be positive if ring is oriented clockwise, otherwise it will be negative.
+     *
+     * Reference:
+     * Robert. G. Chamberlain and William H. Duquette, "Some Algorithms for Polygons on a Sphere", JPL Publication 07-03, Jet Propulsion
+     * Laboratory, Pasadena, CA, June 2007 http://trs-new.jpl.nasa.gov/dspace/handle/2014/40409
+     *
+     */
+    private func ringArea(_ coordinates: [CLLocationCoordinate2D]) -> Double {
+        var p1: CLLocationCoordinate2D
+        var p2: CLLocationCoordinate2D
+        var p3: CLLocationCoordinate2D
+        var lowerIndex: Int
+        var middleIndex: Int
+        var upperIndex: Int
```

Comment:
These six variables are always written inside each iteration of the `for` loop and read immediately thereafter, and never read outside the loop. Move these variables to inside the loop. The original turf-area implementation declares these variables upfront to avoid [hoisting](https://developer.mozilla.org/en-US/docs/Glossary/Hoisting), but it isn’t even necessary to do so in modern, strict-mode JavaScript.